syntax = "proto3";

package eon;

// 计算请求
message ComputationRequest {
    string data_id = 1;
    string operation = 2;
    bytes params = 3;
}

// 计算响应
message ComputationResponse {
    string task_id = 1;
    string status = 2;
}

// 任务状态请求
message TaskStatusRequest {
    string task_id = 1;
}

// 任务状态响应
message TaskStatusResponse {
    string task_id = 1;
    string status = 2;
    float progress = 3;
    string result_id = 4;
    string error = 5;
}

// 计算服务
service ComputationService {
    // 提交计算任务
    rpc SubmitComputation(ComputationRequest) returns (ComputationResponse);
    
    // 获取任务状态
    rpc GetTaskStatus(TaskStatusRequest) returns (TaskStatusResponse);
}